src/generate-elements.ts

```diff
@@ -16,34 +16,59 @@ import * as fs from 'fs';
 import * as jsonschema from 'jsonschema';
 import * as pathLib from 'path';
 
-import {Attribute, Element, ElementLike, ElementMixin, Elements, Property, SourceRange} from './elements-format';
+import {Attribute, Element, ElementLike, ElementMixin, Elements, Namespace, Property, SourceRange} from './elements-format';
+import {Namespace as ResolvedNamespace} from './javascript/namespace';
 import {Document} from './model/document';
+import {Feature} from './model/feature';
 import {Attribute as ResolvedAttribute, Element as ResolvedElement, ElementMixin as ResolvedMixin, Property as ResolvedProperty, SourceRange as ResolvedSourceRange} from './model/model';
 import {Package} from './model/package';
 
 export type ElementOrMixin = ResolvedElement | ResolvedMixin;
 
+export type Filter = (feature: Feature) => boolean;
+
 export function generateElementMetadata(
-    input: Package|Document[], packagePath: string): Elements {
+    input: Package|Document[], packagePath: string, filter?: Filter): Elements {
+  const _filter = filter || ((_: Feature) => true);
+
   let elements: Set<ResolvedElement>;
   let mixins: Set<ResolvedMixin>;
+  let namespaces: Set<ResolvedNamespace>;
 
   if (input instanceof Array) {
```

Comment:
We could get rid of the if/else and reduce a little duplication if the loop were changed to:
```
for (const document of (input instanceof Array ? input : [input])) {
```